packages/core/core/src/AssetGraph.js

```diff
@@ -141,7 +141,16 @@ export default class AssetGraph extends Graph<AssetGraphNode> {
     );
     invariant(entrySpecifierNode.type === 'entry_specifier');
     entrySpecifierNode.correspondingRequest = correspondingRequest;
-    let entryFileNodes = resolved.map(file => nodeFromEntryFile(file));
+    let entryFileNodes = resolved.map(file => {
+      let node = nodeFromEntryFile(file);
+      let existingNode = this.getNode(node.id);
+      if (existingNode) {
+        invariant(existingNode.type === 'entry_file');
+        existingNode.value = file;
+        return existingNode;
+      }
+      return node;
```

Comment:
Should we do this type of thing for every node type? (i.e. replace the value on the existing node rather than replace the whole node). Seems like we've run into this a couple times recently.

Reply:
Yeah, should probably be done in `replaceNodesConnectedTo`

Reply:
Refactored in last commit